frontend/src/metabase/visualizations/components/ChartTooltip/utils.ts

```diff
@@ -18,3 +19,34 @@ export const formatValueForTooltip = ({
     type: "tooltip",
     majorWidth: 0,
   });
+
+export const groupExcessiveTooltipRows = (
+  rows: TooltipRowModel[],
+  maxRows: number,
+) => {
+  if (rows.length <= maxRows) {
+    return rows;
+  }
+
+  const groupStartingFromIndex = maxRows - 1;
+  const result = rows.slice();
+  const rowsToGroup = result.splice(groupStartingFromIndex);
```

Comment:
I find the mutating nature of `.splice()` a little hard to read. I wonder if something like this might be a little clearer?

```suggestion
  const rowsToKeep = rows.slice(0, groupStartingFromIndex);
  const rowsToGroup = result.slice(groupStartingFromIndex);
```